src/plots/cartesian/axes.js

```diff
@@ -1539,40 +1539,26 @@ axes.getTickFormat = function(ax) {
     if(ax.tickformatstops && ax.tickformatstops.length > 0) {
         switch(ax.type) {
             case 'date': {
-                tickstop = ax.tickformatstops.reduce(function(acc, stop) {
-                    if(!isProperStop(ax.dtick, stop.dtickrange, convertToMs)) {
-                        return acc;
-                    }
-                    if(!acc) {
-                        return stop;
-                    } else {
-                        return getRangeWidth(stop.dtickrange, convertToMs) > getRangeWidth(acc.dtickrange, convertToMs) ? stop : acc;
-                    }
-                }, null);
+                tickstop = ax.tickformatstops.find(function(stop) {
```

Comment:
Nice and clean - unfortunately it doesn't look like IE supports `Array.find`, not even in IE11
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find
We've also tended to gravitate toward `for` loops over `Array` methods for performance reasons. Not a big deal here as this array will be fairly small, but it should still help.

Reply:
sorry, just switched from the project where is supported and don't care about IE11 support.
Replaced with `for` loop and fixed tests.